python/ray/util/sgd/v2/backends/backend.py

```diff
@@ -442,15 +453,46 @@ def latest_checkpoint_path(self) -> Optional[Path]:
         else:
             return None
 
+    def _restart(self):
+        if self._num_failures >= self._max_failures:
+            raise RuntimeError("Training has failed even after "
+                               f"{self._num_failures} "
+                               "attempts.")
+        self.worker_group.shutdown()
+        self.worker_group.start()
+        try:
+            self._backend.on_start(self.worker_group, self._backend_config)
```

Comment:
Should this just call  some variant of `self.start()` to ensure that state doesn't diverge?

Reply:
Good point! Made the change